std/typecons.d

```diff
@@ -2387,99 +2387,257 @@ if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArr
     a = new Widget;
 }
 
-// https://issues.dlang.org/show_bug.cgi?id=16054
-@safe unittest
+/// ditto
+struct Rebindable(T)
+if (!is(T == class) && !is(T == interface) && !isDynamicArray!T && !isAssociativeArray!T)
 {
-    Rebindable!(immutable Object) r;
-    static assert(__traits(compiles, r.get()));
-    static assert(!__traits(compiles, &r.get()));
+private:
+    align(T.alignof)
+    static struct Payload
+    {
+        static if (hasIndirections!T)
+        {
+            void[T.sizeof] data;
+        }
+        else
+        {
+            ubyte[T.sizeof] data;
+        }
+    }
+
+    Payload data;
+
+public:
+
+    static if (!__traits(compiles, { T value; }))
+    {
+        @disable this();
+    }
+
+    /**
+     * Constructs a `Rebindable` from a given value.
+     */
+    this(T value) @trusted
+    {
+        set(value);
+    }
+
+    /**
+     * Overwrites the currently stored value with `value`.
+     */
+    void opAssign(this This)(T value) @trusted
+    {
+        clear;
+        set(value);
+    }
+
+    /**
+     * Returns the value currently stored in the `Rebindable`.
+     */
+    T get(this This)() @property @trusted
+    {
+        return *cast(T*) &this.data;
+    }
+
+    ~this() @trusted
+    {
+        clear;
+    }
+
+    ///
+    alias get this;
+
+private:
+
+    void set(this This)(T value)
+    {
+        // As we're escaping a copy of `value`, deliberately leak a copy:
+        static union DontCallDestructor
+        {
+            T value;
+        }
+        DontCallDestructor copy = DontCallDestructor(value);
+        this.data = *cast(Payload*) &copy;
+    }
+
+    void clear(this This)()
+    {
+        // work around reinterpreting cast being impossible in CTFE
+        if (__ctfe)
+        {
+            return;
+        }
+
+        static if (is(T == class) || is(T == interface))
```

Comment:
This branch seems to be dead since the `struct` constraint is `if (!is(T == class) && !is(T == interface) && !isDynamicArray!T && !isAssociativeArray!T)`

Reply:
Er, yeah you're right. In my code, there's one instance that handles everything... 

Removed.